ref/general/application-hardening.adoc

```diff
@@ -0,0 +1,109 @@
+
+== User roles and access
+Understanding the roles and mappings of users is key to securely configuring your application for production use.
+For information about how to map users to roles and groups, see link:/docs/ref/general/#authorization.html[Authorization].
+
+If you don't configure user roles in the `server.xml` file, Open Liberty uses the group name that the application runs under as the role name by default.
+For conflicts in a servlet's role mapping between an application's `web.xml` file and annotations in the application, the config in the `web.xml` file takes precedence.
+If there is a conflict between the `ibm-application-bnd.xmi` and `server.xml` files, then the config int he `server.xml` file takes precedence.
+
+When the role mapping binding information for a protected application isn't provided, the group name and role name must be the same.
+For example, if the role name is `manager`, then a user who belongs to a `manager` group has access to that resource.
+This applies only when application binding information isn't specified for the application in the `ibm-application-bnd.xmi` or `server.xml` files
+
+== Web server document root
+WAR files contain application code and sensitive information.
+Only some of that information is web-servable content, so the web server document root should not be set to the WAR root.
+If the web server document root is set to the WAR root, the web server serves up all the contents of the WAR file without interpretation, meaning that code, raw JSPs, and more are served to your users.
+
+You should also never place sensitive information in the WAR root.
+The web container serves any files that are found in the root of the WAR file, which is okay if the root contains only servable content.
+However, because the web container serves files from the WAR root, you must never place any content that users shouldn't see in the WAR root.
+For example, don't put property files, class files, or other important information in there.
+If you must place such information in the WAR file, place it in the `WEB-INF` directory, as permitted by the servlet specification.
+Information in this directory is never served by the web container.
+
+== Secure servlets
+Servlets can be served by class name or with a normal URL alias.
+By default, Open Liberty does not allow you to server servlets by class name.
+Instead of defining a mapping for each servlet, a single generic URL (such as `/servlet`) serves all servlets.
+The component of the path after the base is assumed to be the class name for the servlet.
+For example, `/servlet/com.ibm.sample.MyServlet` refers to the `com.ibm.sample.MyServlet` servlet class.
+
+You can serve servlets by class name by setting the `serveServletsByClassnameEnabled` property to `true` in the `ibm-web-ext.xmi` file.
+For production purposes, don't enable this property.
+Enabling it makes it possible for anyone that knows the class name of any servlet to directly invoke it.
```

Comment:
For "Enabling it makes it possible", I think you should maybe mention the property name again here to avoid confusion so maybe change that to "Enabling the `serveServletsByClassnameEnabled` property makes it possible"

Reply:
Updated